### PR TITLE
EnergyDynamics propagated from MonoLayer to MonoLayerDyn in MonoLayer.mo

### PR DESCRIPTION
**Pull request by GlennReynders**
Fixed issue461

- EnergyDynamics propagated to monoLayer
- Parameter for EnergyDynamics included for BoundaryWall.mo  Added conditional declaration for use_T_in to make sure free initial conditions are used.

Passed all unit tests

### CODE REVIEW

File: IDEAS/Buildings/Components/BoundaryWall.mo [hunk 1]
```diff
@@ -2,8 +2,11 @@ within IDEAS.Buildings.Components;
 model BoundaryWall "Opaque wall with boundary conditions"
   extends IDEAS.Buildings.Components.Interfaces.PartialOpaqueSurface(
      final QTra_design=U_value*AWall*(273.15 + 21 - TRef_a),
-     dT_nominal_a=-1);
-
+     dT_nominal_a=-1,
+    layMul(energyDynamics=energyDynamics));
+  parameter Modelica.Fluid.Types.Dynamics energyDynamics= if use_T_in then Modelica.Fluid.Types.Dynamics.DynamicFreeInitial else  Modelica.Fluid.Types.Dynamics.FixedInitial
```
Comment (Mathadon) [inline IDEAS/Buildings/Components/BoundaryWall.mo]: This default value can be changed by the user and therefore it would be better to add the `if` on line 6 such that:
```
layMul(energyDynamics=if use_T_in then Modelica.Fluid.Types.Dynamics.DynamicFreeInitial else  energyDynamics));
parameter Modelica.Fluid.Types.Dynamics energyDynamics= Modelica.Fluid.Types.Dynamics.FixedInitial
```

**Comment by Mathadon** [comment]
This is superseded by #469
